File: api/src/endpoint/migration/migrate_github_repositories.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::migration::MigrationResource};

pub struct MigrateGitHubRepositories;

impl Endpoint for MigrateGitHubRepositories {
    const PATH: &'static str = "/migration/github/{installation_id}/migrate";
    const METHOD: http::Method = http::Method::POST;

    type Request = MigrateGitHubRepositoriesRequest;
    type Response = MigrateGitHubRepositoriesResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct MigrateGitHubRepositoriesRequest {
    pub owner: String,
    pub owner_type: String,
    pub repositories: Vec<String>,
}

pub type MigrateGitHubRepositoriesResponse = MigrationResource;

File: gitdot-api-ts/src/endpoint/migration/github/migrate-github-repositories.ts
import { z } from "zod";

import { MigrationResource } from "../../../resource";

export const MigrateGitHubRepositoriesRequest = z.object({
  origin: z.string(),
  origin_type: z.string(),
  destination: z.string(),
  destination_type: z.string(),
  repositories: z.array(z.string()),
});
export type MigrateGitHubRepositoriesRequest = z.infer<
  typeof MigrateGitHubRepositoriesRequest
>;

export const MigrateGitHubRepositories = {
  path: "/migration/github/{installation_id}/migrate",
  method: "POST",
  request: MigrateGitHubRepositoriesRequest,
  response: MigrationResource,
} as const;
export type MigrateGitHubRepositories = typeof MigrateGitHubRepositories;
